sql/pipelines/src/main/scala/org/apache/spark/sql/pipelines/graph/GraphValidations.scala

```diff
@@ -55,6 +55,75 @@ trait GraphValidations extends Logging {
     multiQueryTables
   }
 
+  protected[graph] def validateFlowStreamingness(): Unit = {
+    flowsTo.foreach { case (destTableIdentifier, flows) =>
+      val destTableOpt = table.get(destTableIdentifier)
+
+      // If the destination identifier does not correspond to a table, it must be a view.
+      val destViewOpt = destTableOpt.fold(view.get(destTableIdentifier))(_ => None)
+
+      flows.foreach {
```

Comment:
Nitpick you can feel free to ignore: this is pretty heavily nested. Could remove a level of nesting with:

```
val resolvedFlows: Seq[ResolvedFlow] = flows.collect { case rf: ResolvedFlow => rf }
```